packages/alfa-rectangle/src/rectangle.ts

```diff
@@ -0,0 +1,157 @@
+import { Equatable } from "@siteimprove/alfa-equatable";
+import { Hash, Hashable } from "@siteimprove/alfa-hash";
+import { Serializable } from "@siteimprove/alfa-json";
+
+import * as json from "@siteimprove/alfa-json";
+
+const { max, min } = Math;
+
+/**
+ * @public
+ */
+export class Rectangle
+  implements Equatable, Hashable, Serializable<Rectangle.JSON>
+{
+  public static of(
+    x: number,
+    y: number,
+    width: number,
+    height: number
+  ): Rectangle {
+    return new Rectangle(x, y, width, height);
+  }
+
+  private readonly _x: number;
+  private readonly _y: number;
+  private readonly _width: number;
+  private readonly _height: number;
+
+  private constructor(x: number, y: number, width: number, height: number) {
+    this._x = x;
+    this._y = y;
+    this._width = width;
+    this._height = height;
+  }
+
+  get width() {
```

Comment:
**Chore:** we usually have both explicit `public` modifier, and explicit return type.
```suggestion
  public get width(): number {
```